File: InvotraxApp/backend/InvotraxApp/src/main/java/dev/lkeleti/invotraxapp/dto/CreateProductCommand.java
package dev.lkeleti.invotraxapp.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class CreateProductCommand {
    private String name;
    private String sku;
    private String description;
    private String unit;
    private Long categoryId;
    private Long manufacturerId;
    private BigDecimal netPurchasePrice;
    private BigDecimal grossPurchasePrice;
    private BigDecimal netSellingPrice;
    private BigDecimal grossSellingPrice;
    private Integer warrantyPeriodMonths;
    private int stockQuantity;
    private boolean serialNumberRequired;
    private Long vatRateId;
    private Long productTypeId;
    private List<BarcodeDto> barcodes;
    private List<SerialNumberDto> serialNumbers;
}

File: InvotraxApp/frontend/invotrax/src/app/models/create-product-command.ts
import { BarcodeDto } from "./barcode.dto";
import { SerialNumberDto } from "./serial-number.dto";

export interface CreateProductCommand {
    name: string;
    sku: string;
    description: string;
    productTypeId: number;
    categoryId: number;
    manufacturerId: number;
    netSellingPrice: number;
    grossSellingPrice: number;
    warrantyPeriodMonths: number;
    serialNumberRequired: boolean;
    vatRateId: number;
    unit: string;
    barcodes: BarcodeDto[];
    serialNumbers: SerialNumberDto[];
}